src/components/Table/TableBase.tsx

```diff
@@ -404,16 +407,50 @@ export class TableBase<T> extends React.Component<
 		const identifier = item[rowKey];
 		const newIsChecked = !this.getCheckedRowIdentifiers().has(identifier);
 
-		const oldCheckedItems = this.props.checkedItems || this.state.checkedItems;
-		const checkedItems = newIsChecked
-			? oldCheckedItems.concat(item)
-			: (oldCheckedItems ?? []).filter((x) => x[rowKey] !== identifier);
+		let checkedItems = this.props.checkedItems || this.state.checkedItems;
+
+		let keysToFind = [identifier];
+		if (
+			this.state.lastSelected !== null &&
+			(e.nativeEvent as MouseEvent).shiftKey
+		) {
+			keysToFind.push(this.state.lastSelected);
+		}
+		let index = 0;
+		while (keysToFind.length) {
```

Comment:
Let's make this a `for` so that it's bounded and clear what it does
```suggestion
		for (let index = 0; index < data.length; index++) {
```
and then add at the very end `if (keysToFind.length) { break; }` to stop once we are done finding all keys.

Reply:
Actually going to go with `for (let index = 0; index < data.length && !!keysToFind.length; index++)`